extension/src/openvic-extension/classes/GUINode.cpp

```diff
@@ -196,3 +233,125 @@ String GUINode::format_province_name(String const& province_identifier) {
+		Vector2i new_img_size = img_size.max(total_size);
+		if (new_img_size != img_size) {
+			img->crop(new_img_size.x, new_img_size.y);
+		}
+		img->blend_rect(texture_img, texture_img->get_used_rect(), control->get_position());
+	}
+	ERR_FAIL_COND_V(img->is_empty(), false);
+	return true;
+}
+
+void GUINode::update_click_mask() {
+	static constexpr real_t max_real = std::numeric_limits<real_t>::max();
+	static const Point2 max_point { max_real, max_real };
+	if (_mask_controls.size() == 0) {
+		return;
+	}
+
+	if (_click_mask.is_null()) {
+		_click_mask.instantiate();
+	}
+	Ref<Image> img;
+	img.instantiate();
+	Vector2 size = get_size();
+	img->create(size.x, size.y, false, Image::Format::FORMAT_RGBA8);
+	Point2 highest_position = { max_real, max_real };
+	for (int index = 0; index < _mask_controls.size(); index++) {
+		if (!_update_click_mask_for(img, index)) {
+			continue;
+		}
+		Vector2 screen_pos = _mask_controls[index]->get_screen_position();
+		if (screen_pos.x < highest_position.x && screen_pos.y < highest_position.y) {
+			highest_position = screen_pos;
+		}
+	}
+	ERR_FAIL_COND(img.is_null());
+	ERR_FAIL_COND(highest_position == max_point);
+	_texture_region = Rect2(Point2(), img->get_size());
+	_position_rect = Rect2(highest_position, _texture_region.get_size());
+	_click_mask->create_from_image_alpha(img);
+	queue_redraw();
+	update_minimum_size();
+}
+
+void GUINode::set_click_mask_from_nodepaths(TypedArray<NodePath> const& paths) {
+	// TODO: Update to use https://github.com/godotengine/godot/pull/90916
+	// for(godot::Control* control : _mask_controls) {
+	// 	control->set_mouse_filter(Control::MouseFilter::MOUSE_FILTER_STOP);
+	// }
+	_mask_controls.clear();
+	for (int index = 0; index < paths.size(); index++) {
+		Control* control = _cast_node<Control>(get_node_internal(paths[index]));
+		ERR_CONTINUE(control == nullptr);
+		control->set_mouse_filter(Control::MouseFilter::MOUSE_FILTER_IGNORE);
+		_mask_controls.push_back(control);
+	}
+	update_click_mask();
+}
+
+bool GUINode::_has_point(godot::Vector2 const& p_point) const {
+	if (!_click_mask.is_valid()) {
+		return Control::_has_point(p_point);
+	}
+
+	Point2 point = p_point;
+	Rect2 rect;
+	Size2 mask_size = _click_mask->get_size();
+
+	if (!_position_rect.has_area()) {
+		rect.size = mask_size;
+	} else {
+		// we need to transform the point from our scaled / translated image back to our mask image
+		Point2 ofs = _position_rect.position;
+		Size2 scale = mask_size / _position_rect.size;
+
+		// offset and scale the new point position to adjust it to the bitmask size
+		point -= ofs;
+		point *= scale;
+
+		// finally, we need to check if the point is inside a rectangle with a position >= 0,0 and a size <= mask_size
+		rect.position = Point2(MAX(0, _texture_region.position.x), MAX(0, _texture_region.position.y));
+		rect.size = Size2(MIN(mask_size.x, _texture_region.size.x), MIN(mask_size.y, _texture_region.size.y));
```

Comment:
```suggestion
		rect.position = _texture_region.position.max(Point2 {});
		rect.size = mask_size.min(_texture_region.size);
```
Could use these nice helper functions.